File: ReactTCCCLogic/DataPoints/DataPointDefinitions.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;

namespace ReactFrameworkLogic.DataPoints
{
    public class DataPointDefinitions
    {
        public readonly static DataPointDescriptor CASE_NAME = new DataPointDescriptor("CN", "CN", Encodings.String);
        public readonly static DataPointDescriptor PATIENT_SURNAME = new DataPointDescriptor("PSN", "PSN", Encodings.String);
        public readonly static DataPointDescriptor PATIENT_FORENAME = new DataPointDescriptor("PFN", "PFN", Encodings.String);

        public static Dictionary<string, DataPointDescriptor> DATAPOINTDEFINITIONS = (new KeyValuePair<string, DataPointDescriptor>[]
        {
            new KeyValuePair<string, DataPointDescriptor>(PATIENT_SURNAME.DataPointName,PATIENT_SURNAME),
            new KeyValuePair<string, DataPointDescriptor>(PATIENT_FORENAME.DataPointName, PATIENT_FORENAME),
            new KeyValuePair<string, DataPointDescriptor>(CASE_NAME.DataPointName, CASE_NAME)

        }).ToDictionary(k => k.Key, v => v.Value);
    }
}

File: TCCCReactWebApp/ClientApp/store/DataPointDefinitions.ts
import DatapointDescriptor, { EncodingTypes } from "./DataPointDescriptor";

export default class DatapointDefinitions {
    // this is the only one we should have in EVERY implementation.
    public static readonly CONTAINERNAME = new DatapointDescriptor("CN", "CN", "Case Name");

    // TCCC
    public static readonly PATIENTFORENAME = new DatapointDescriptor("PFN", "PFN", "Forename");
    public static readonly PATIENTSURNAME = new DatapointDescriptor("PSN", "PSN", "Surname");
    public static readonly PATIENTDATEOFBIRTH = new DatapointDescriptor("PDOB", "PDOB", "Date of Birth", EncodingTypes.Date);

    // provide a lookup function for our datapoints
    public static Lookup: { [key: string]: DatapointDescriptor } = [
        DatapointDefinitions.CONTAINERNAME,
        DatapointDefinitions.PATIENTFORENAME, DatapointDefinitions.PATIENTSURNAME, DatapointDefinitions.PATIENTDATEOFBIRTH
        ]
        .reduce((p: { [key: string]: DatapointDescriptor }, c: DatapointDescriptor) => { p[c.dataPointName] = c; return p; }, {});
}